src/VisualStudio/Core/Next/Remote/ServiceHubRemoteHostClient.cs

```diff
@@ -136,6 +164,40 @@ private void OnRpcDisconnected(object sender, JsonRpcDisconnectedEventArgs e)
             Disconnected();
         }
 
+        /// <summary>
+        /// call <paramref name="funcAsync"/> and retry up to <paramref name="timeout"/> if the call throws
+        /// <typeparamref name="TException"/>. any other exception from the call won't be handled here.
+        /// </summary>
+        private static async Task<TResult> RetryRemoteCallAsync<TException, TResult>(
+            Func<Task<TResult>> funcAsync,
+            TimeSpan timeout,
+            CancellationToken cancellationToken) where TException : Exception
+        {
+            const int retry_delayInMS = 50;
+
+            var start = DateTime.UtcNow;
+            while (DateTime.UtcNow - start < timeout)
```

Comment:
Should you be using `Stopwatch` here instead of doing manual arithmetic on DateTime?

Reply:
to prevent a bug you mean? I dont need high precision for timeout. just roughly timeout. so stopwatch sounds like an overkill considering this get called many many times?

Reply:
I prefer Stopwatch whenever I need timer like functionality since it reads easier and I feel like I have less chance of getting my arithmetic wrong through a simple authoring error.
Also, I assume that if the user changed his system time during the interval, that if that caused my code to malfunction it would be a bug in Stopwatch rather than my own datetime arithmetic logic.

Reply:
"user changed his system time", ah another case where I didn't think of. yes. DateTime will get affected by it.

Reply:
At least you used `UtcNow` instead of `Now`, so I think you're defending against changes to time *zone* and DST.

Reply:
looks like Stopwatch is also using DateTime if higher precision counter is not available to the machine... so it is also not 100% safe from time change..